### PR TITLE
Namespace support

### PR DESCRIPTION
**Pull request by zikicm**
I've added support for namespaces. In the end I did not want to use classes in order to make this file independent from other libraries. Please check if naming and location of the file are appropriate. This implementation of feature #6.

### CODE REVIEW

File: libs/define.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+// In order to make our code structured, modular and reusable
```
Comment (comanjoni) [inline libs/define.js]: My idea for libs was to put folders in it and source code in folders, not directly in libs.
Let we create folder (lib in libs) called util (or lang, or core, or find proper name) for define.js.

File: libs/define.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+(function(window){
+
+	// Maps namespace to module
+	var _moduleMap = {};
+
+	// Fetches exports from module based on module's namespace
+	var require = function(namespace) {
+
+		if (!_moduleMap[namespace]) {
+			throw "Module " + namespace + " not defined!";
```
Comment (comanjoni) [inline libs/define.js]: Let we throw error not a string. Error should keep string in it.

File: libs/define.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+		return _moduleMap[namespace].exports;
+
+	};
+
+	// Global function used for defining of namespaces
+	window.define = function(namespace, moduleFunc) {
+
+		if (_moduleMap[namespace]) {
+			throw "Module " + namespace + " already defined!";
```
Comment (comanjoni) [inline libs/define.js]: The same for this error...

**Comment by comanjoni** [comment]
This is ok. Just resolve comments and we don't need 2nd iteration.

File: libs/define.js (hunk 3 above)
Comment (zikicm) [inline libs/define.js]: :+1:

File: libs/define.js (hunk 2 above)
Comment (zikicm) [inline libs/define.js]: :+1:

**Comment by zikicm** [comment]
Comments resolved and merged into master via bd3a5e7dff389e92a595c8f2564201f5f92cbcb0.

**Pull request closed without merging**